test/e2e/antctl_test.go

```diff
@@ -0,0 +1,144 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// antctlOutput is a helper function for logging antctl outputs.
+func antctlOutput(stdout, stderr string, tb testing.TB) {
+	tb.Logf("antctl stdout:\n%s", stdout)
+	tb.Logf("antctl stderr:\n%s", stderr)
+}
+
+// runAntctl runs antctl commands on antrea pods, the controller, or agents. It
+// always runs the commands with verbose flag enabled.
+func runAntctl(podName string, subCMDs []string, data *TestData, tb testing.TB) (string, string, error) {
+	var containerName string
+	if strings.Contains(podName, "agent") {
+		containerName = "antrea-agent"
+	} else {
+		containerName = "antrea-controller"
+	}
+	cmds := []string{"antctl", "-v"}
+	stdout, stderr, err := data.runCommandFromPod(antreaNamespace, podName, containerName, append(cmds, subCMDs...))
+	antctlOutput(stdout, stderr, tb)
+	return stdout, stderr, err
+}
+
+// TestAntctlAgentLocalAccess ensures antctl is accessible in agent pod.
+func TestAntctlAgentLocalAccess(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	podName, err := data.getAntreaPodOnNode(masterNodeName())
+	if err != nil {
+		t.Fatalf("Error when getting antrea-agent pod name: %v", err)
+	}
+	if _, _, err := runAntctl(podName, []string{"version"}, data, t); err != nil {
+		t.Fatalf("Error when running `antctl version` from %s: %v", podName, err)
+	}
+}
+
+// TestAntctlControllerLocalAccess ensures antctl is accessible in controller pod.
+func TestAntctlControllerLocalAccess(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	podName, err := data.getAntreaController()
+	if err != nil {
+		t.Fatalf("Error when getting antrea-controller pod name: %v", err)
+	}
+	if _, _, err := runAntctl(podName, []string{"version"}, data, t); err != nil {
+		t.Fatalf("Error when running `antctl version` from %s: %v", podName, err)
+	}
+}
+
+// TestAntctlControllerRemoteAccess ensures antctl is able to be run outside of
+// the kubernetes cluster. It uses the antctl client binary copied from the controller
+// pod.
+func TestAntctlControllerRemoteAccess(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	podName, err := data.getAntreaController()
+	assert.Nil(t, err, "Error when retrieving antrea controller pod name")
```

Comment:
here and in other places below, `require` may be more appropriate than `assert`. If we cannot retrieve the Pod name here, we probably want to interrupt the test.

Reply:
Thanks, I expected `assert` just works like it does in other languages. Updated.